generators/generator-bot-adaptive/tests/suites/js-functions.test.js

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+const helpers = require('yeoman-test');
+const path = require('path');
+const TemplateFileReader = require('../templateFileReader');
+
+const botName = 'JavaScriptFunctions';
+const generatorPath = path.join(__dirname, '..', '..', 'generators', 'app');
+const integration = 'functions';
+const platform = 'js';
+const reader = new TemplateFileReader(path.join(generatorPath, 'templates'));
+
+let runResult;
+
+before(async function () {
+  runResult = await helpers
+    .create(generatorPath)
+    .withArguments([botName])
+    .withOptions({ platform, integration })
+    .run();
+});
+
+after(function () {
+  if (runResult) {
+    runResult.restore();
+  }
+});
+
+it(`should create file ${path.join(botName, 'appsettings.json')}`, function () {
+  const filePath = path.join(botName, 'appsettings.json');
+  const content = reader.getJsonFileContent(
+    path.join('assets', 'appsettings.json'),
+    {
+      luis: {
+        name: botName,
+      },
+      runtime: {
+        command: `npm run dev --`,
+        key: `adaptive-runtime-${platform}-${integration}`,
+      },
+    }
+  );
+
+  runResult.assertJsonFileContent(filePath, content);
+});
+
+it(`should create file ${path.join(botName, 'host.json')}`, function () {
+  const filePath = path.join(botName, 'host.json');
+  const content = reader.getFileContent(
+    path.join('js', 'functions', 'host.json')
+  );
+
+  runResult.assertFileContent(filePath, content);
+});
+
+it(`should create file ${path.join(botName, 'index.js')}`, function () {
+  const filePath = path.join(botName, 'index.js');
+  const content = reader.getTemplateFileContent(
+    path.join('js', 'functions', 'index.js'),
+    {
+      settingsDirectory: 'process.cwd()',
+    }
+  );
+
+  runResult.assertFileContent(filePath, content);
+});
+
+it(`should create file ${path.join(
+  botName,
+  `${botName}.botproj`
+)}`, function () {
+  const filePath = path.join(botName, `${botName}.botproj`);
+  const content = reader.getJsonFileContent(
+    path.join('assets', 'botName.botproj'),
+    {
+      name: botName,
+    }
+  );
+
+  runResult.assertJsonFileContent(filePath, content);
+});
+
+it(`should create file ${path.join(
+  botName,
+  'local.settings.json'
+)}`, function () {
+  const filePath = path.join(botName, 'local.settings.json');
+  const content = reader.getFileContent(
+    path.join('js', 'functions', 'local.settings.json')
+  );
+
+  runResult.assertFileContent(filePath, content);
+});
+
+it(`should create file ${path.join(botName, 'package.json')}`, function () {
+  const filePath = path.join(botName, 'package.json');
+  const content = {
+    name: botName,
+    private: true,
+    scripts: {
+      dev: 'cross-env NODE_ENV=dev func start',
```

Comment:
I wonder if we should be a bit less strict about these expectations - I suppose we can adjust this if it gets annoying having to update commands or dependency versions in two places.

Reply:
Spoke offline - going to leave as is for now. If this becomes problematic, it can be addressed in the future.